pkg/util/managedcluster/managedcluster.go

```diff
@@ -43,30 +43,19 @@ func getRestConfigFromV1Config(kc *v1.Config) (*rest.Config, error) {
 	return kubeconfig.ClientConfig()
 }
 
-// ClientsetFromV1Config takes a v1 config and returns a Clientset
-func ClientsetFromV1Config(config *v1.Config) (*kubernetes.Clientset, error) {
-	restconfig, err := getRestConfigFromV1Config(config)
-	if err != nil {
-		return nil, err
-	}
-
-	return kubernetes.NewForConfig(restconfig)
-}
-
-// WaitForHealthz takes a context, v1 config.
+// WaitForHealthz takes a context, a OpenShiftManagedCluster, and a logrus.Entry
 // It waits for the cluster to respond to healthz requests.
-func WaitForHealthz(ctx context.Context, config *v1.Config) error {
-	restconfig, err := getRestConfigFromV1Config(config)
+func WaitForHealthz(ctx context.Context, cs *api.OpenShiftManagedCluster, log *logrus.Entry) error {
+	restconfig, err := RestConfigFromV1Config(cs.Config.AdminKubeconfig)
 	if err != nil {
 		return err
 	}
-
 	t, err := rest.TransportFor(restconfig)
 	if err != nil {
 		return err
 	}
 
 	// Wait for the healthz to be 200 status
-	_, err = wait.ForHTTPStatusOk(ctx, t, restconfig.Host+"/healthz")
+	_, err = wait.ForHTTPStatusOk(ctx, t, restconfig.Host+"/healthz", log)
```

Comment:
@jim-minter, our overarching goal here was to remove the restconfig bits.  I was unsuccessful at doing so because I was unable to create a proper transport required to verify the x509 certificate without it.  This caused the createorupdate.go to fail each time.

Is there another way to create the proper transport from the `OpenShiftManagedCluster` object and avoid creating the `restconfig`?  If that's possible then we can do away with the line 49-52 and somehow create the proper transport.

Reply:
Does this work?

```go
pool := x509.NewCertPool()
pool.AddCert(cs.Config.Certificates.Ca.Cert)

t := &http.Transport{
	TLSClientConfig: &tls.Config{
		RootCAs: pool,
	},
}
```

Reply:
Will test with this.  I tried a few iterations of something similar.